docs/topics/Gateway.md

```diff
@@ -786,6 +801,49 @@ Sent when a channel is updated. The inner payload is a [channel](#DOCS_RESOURCES
 
 Sent when a channel relevant to the current user is deleted. The inner payload is a [channel](#DOCS_RESOURCES_CHANNEL/channel-object) object.
 
+#### Thread Create
+
+Sent when a thread is created, relevant to the current user, or when the current user is added to a thread. The inner payload is a [channel](#DOCS_RESOURCES_CHANNEL/channel-object) object.  When being added to an existing private thread, includes a [thread member](#DOCS_RESOURCES_CHANNEL/thread-member-object) object.
+
+#### Thread Update
+
+Sent when a thread is updated. The inner payload is a [channel](#DOCS_RESOURCES_CHANNEL/channel-object) object. This is not sent when the field `last_message_id` is altered. To keep track of the last_message_id changes, you must listen for [Message Create](#DOCS_TOPICS_GATEWAY/message-create) events.
+
+#### Thread Delete
+
+Sent when a thread relevant to the current user is deleted. The inner payload is a subset of the [channel](#DOCS_RESOURCES_CHANNEL/channel-object) object, containing just the `id`, `guild_id`, `parent_id`, and `type` fields.
+
+#### Thread List Sync
+
+Sent when the current user _gains_ access to a channel.
+
+###### Thread List Sync Event Fields
+
+| Field        | Type                                                                           | Description                                                                                                                               |
+|--------------|--------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------|
+| guild_id     | snowflake                                                                      | the id of the guild                                                                                                                       |
+| channel_ids? | array of snowflakes                                                            | the parent channel ids whose threads are being synced.  If omitted, then threads were synced for the entire guild                         |
+| threads      | array of [channel](#DOCS_RESOURCES_CHANNEL/channel-object) objects             | all active threads in the given channels that the current user can access                                                                 |
+| members      | array of [thread member](#DOCS_RESOURCES_CHANNEL/thread-member-object) objects | all thread member objects from the synced threads for the current user, indicating which threads the current user has been added to       |
```

Comment:
To be clear, the `channel_ids` and `members` information is available in a nested format inside of the `threads` array, correct? (`threads[*].parentId` and `threads[*].member`)

Are these root-level properties just for making certain access patterns easier or is there a specific reason for them?

Reply:
Check out the `Thread Membership` section and if you still have questions or concerns about it, let me know, but it talks about this

Reply:
Also `channel_ids` can't be inferred from the returned threads.
I expect the algorithm bots should use here is:
- Remove all threads that have a parent_id that is in the channel_ids set
- Add all threads from the threads array
Because a channel might not have any active threads in it, so we're signaling that to you here as well!